backend/src/test/java/com/ambacam/rest/autorites/AutoritesResourceIT.java

```diff
@@ -0,0 +1,171 @@
+	public void createPosteNull() {
+		Autorite create = buildAutorite();
+		create.setPoste(null);
+		given().contentType(ContentType.JSON).body(create).log().body().post(ApiConstants.AUTORITE_COLLECTION)
+				.then().log().body().statusCode(400);
+	}
+
+	@Test
+	public void createMemeNom() {
+		Autorite create = buildAutorite();
+		create.setNom(autorite1.getNom());
+		given().contentType(ContentType.JSON).body(create).log().body().post(ApiConstants.AUTORITE_COLLECTION)
+				.then().log().body().statusCode(400);
+	}
+
+	@Test
+	public void list() {
+		given().get(ApiConstants.AUTORITE_COLLECTION).then().log().body().statusCode(200)
+				.body("size()", is(equalTo(2)))
+				.body("id", containsInAnyOrder(autorite1.getId().intValue(), autorite2.getId().intValue()));
+	}
+
+	@Test
+	public void get() {
+		given().get(ApiConstants.AUTORITE_ITEM, autorite2.getId()).then().log().body().statusCode(200)
+				.body("id", is(equalTo(autorite2.getId().intValue())))
+				.body("nom", is(equalTo(autorite2.getNom())));
+	}
+
+	@Test
+	public void getNotFound() {
+		given().get(ApiConstants.AUTORITE_ITEM, random.nextLong()).then().statusCode(404);
+	}
+
+	@Test
+	public void delete() {
+		given().delete(ApiConstants.AUTORITE_ITEM, autorite1.getId()).then().statusCode(200);
+
+		// check that the autorite has been deleted
+		Autorite actual = repository.findOne(autorite1.getId());
+		assertThat(actual, is(nullValue()));
+
+	}
+
+	@Test
+	public void deleteNotFound() {
+		given().delete(ApiConstants.AUTORITE_ITEM, random.nextLong()).then().statusCode(404);
+	}
+
+	@Test
+	public void update() {
+		Autorite update = buildAutorite();
+		given().contentType(ContentType.JSON).body(update).put(ApiConstants.AUTORITE_ITEM, autorite2.getId())
+				.then().log().body().statusCode(200);
+
+		// check that the autorite has been saved
+		Autorite actual = repository.findOne(autorite2.getId());
+		assertThat(actual.getId(), is(equalTo(autorite2.getId())));
+		assertThat(actual.getNom(), is((equalTo(update.getNom()))));
+
+	}
+
+	@Test
+	public void updateNotFound() {
+		Autorite update = buildAutorite();
+		given().contentType(ContentType.JSON).body(update).put(ApiConstants.AUTORITE_ITEM, random.nextLong())
+				.then().log().body().statusCode(404);
+	}
+
+	@Test
+	public void updateMemeNom() {
+		Autorite update = buildAutorite();
+		update.setNom(autorite1.getNom());
+		given().contentType(ContentType.JSON).body(update).log().body()
+				.put(ApiConstants.AUTORITE_ITEM, autorite1.getId()).then().log().body().statusCode(200);
+
+		// check that the autorite has been saved
+		Autorite actual = repository.findOne(autorite1.getId());
+		assertThat(actual.getNom(), is((equalTo(update.getNom()))));
+	}
+
+	@Test
+	public void updateNomExists() {
+		Autorite update = buildAutorite();
+		update.setNom(autorite2.getNom());
+		given().contentType(ContentType.JSON).body(update).log().body()
+				.put(ApiConstants.AUTORITE_ITEM, autorite1.getId()).then().log().body().statusCode(400);
```

Comment:
This test should not give the status 400 because the name is not unique

Reply:
I also removed this test, since there is no constraint of unicity on property 'nom'